### ISSUE TITLE
Can't Unlock or Unenroll Device

### ISSUE THREAD

**Issue by GraemeAGBarr** [google/loaner#53]
Device would unlock to be used again, or unenroll to allow re enrollment 

Fails with "ERROR Precondition Failed" Message

Go into console find device, click 3 dots, choose unlock, say yes to are you sure prompt, error appears and device remains locked.
Same for unenroll.

This is the same admin account that was used to setup the loaner in the 1st instance, so permissions are unchanged.

**Comment by helfrichmichael** [comment]
Are there any errors in the Developer Tools?

https://developers.google.com/web/tools/chrome-devtools/open

**Comment by GraemeAGBarr** [comment]
Resolved by Google dev team.

**Comment by ericwhiteau** [comment]
What was the resolution.  I have the same issue.

I have one site which unlocks successfully.  all good.

we have a second instance  and two chromebooks are locked.  and give the above error when you  run the unlock function?

**Comment by ianmagnone** [comment]
I have this same issue and would love to know what the Google Dev team did to fix it.

**Comment by helfrichmichael** [comment]
@ianmagnone I resolved this issue IIRC for @ericwhiteau and @GraemeAGBarr , but unfortunately I don't recall what the issue was.

Can you confirm if you have any issues in the JS console or in App Engine's log viewer when you make this request?

**Comment by ericwhiteau** [comment]
hello all,
From memory, I had to manually change the entity in Datastore - Device    Locked -> false

**Comment by templesnr** [comment]
one trick I used was to enroll the device again. I found that on some
occasions an admin unlocked the device from the admin console rather than
the loaner app, which caused the error. I found that removing the device
and then adding it again cleared the issue.

**Comment by templesnr** [comment]
p.s. I've also used the method that Eric described too

**Comment by helfrichmichael** [comment]
@templesnr and @ericwhiteau you both rock ❤️ . 

Thanks for helping out here and hopefully that can unblock/resolve the issue for @ianmagnone .

**Comment by ianmagnone** [comment]
I spent some time on this the other day when I was in the office. Between a combination of making sure the device was happy in Google Admin, and unenrolling/re-enrolling in Grab-n-Go I was able to get them working again.

It would be helpful if any error messages in the app portal for the admin would give more info than "Precondition failed" to help narrow down what might be wrong. Is that something that is possible?
